dgraham/Ka-Block
# Add macOS app

This PR introduces a work in progress native macOS app with Content Blocker extension. The code is very similar to the iOS app.

To test:

* Checkout branch and open Xcode.
* Select the "Ka-Block! macOS" build scheme.

<img width="307" alt="screen shot 2017-08-31 at 8 10 57 pm" src="https://user-images.githubusercontent.com/137/29953884-8486c17e-8e88-11e7-9f28-ee3bc355d56e.png">

* Build it, you don't have to run the shell app.
* Open Safari's Develop menu and "Allow Unsigned Extensions".

<img width="478" alt="screen shot 2017-08-31 at 8 08 34 pm" src="https://user-images.githubusercontent.com/137/29953907-a3817902-8e88-11e7-89ee-3cd551759281.png">

* Enable the Safari Extension.

<img width="888" alt="screen shot 2017-08-31 at 8 08 58 pm" src="https://user-images.githubusercontent.com/137/29953921-b954d8d2-8e88-11e7-9948-95df685a0fd5.png">

* Verify it works!

![screen shot 2017-08-31 at 8 09 09 pm](https://user-images.githubusercontent.com/137/29953926-c7cfc82c-8e88-11e7-9159-81f0424a2b54.png)

**Remaining Work**

I think this PR can be merged and we can continue to work on the macOS shell application. Some remaining work:

* Design macOS icon. Update `.appiconset`. Maybe this resource could be shared between targets.
* Design shell application launch screen. Maybe it could look similar to the iOS version.
* Investigate sharing `ContentBlockerRequestHandler` class between targets.
